wordpress-mobile/WordPress-iOS
# WP.com Plans: Result screen

Fixes https://github.com/wordpress-mobile/WordPress-iOS/issues/20690

Integrates the final purchase result screen into the WP.com plan purchase flow.

## Visuals

| New screen | Entire flow |
| - | - |
| <img alt="Screenshot 2023-06-01 at 18 43 07" width="400" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1898325/370f512b-23da-437d-88b1-69f949a1208b"> | <video src="https://github.com/wordpress-mobile/WordPress-iOS/assets/1898325/eca98d21-edc0-4022-b3ff-66995f81b73f"> |





## Testing steps

On iPhone:

1. Log in using a WP.com account
2. Enable `Free to Paid Plans Dashboard Card` feature flag
3. Tap the `Free domain with an annual plan` card
4. Follow the steps to purchase a WP.com plan 
5. Expect to see the new "All ready to go!" screen at the end with the domain name in bold
6. Tap done to close the flow

Known issues:
- https://github.com/wordpress-mobile/WordPress-iOS/issues/20789
- https://github.com/wordpress-mobile/WordPress-iOS/issues/20790
- ~~https://github.com/wordpress-mobile/WordPress-iOS/issues/20791~~

I think these can be handled in a follow-up PR so I created separate issues for them, but happy to reconsider if you think they should be done on this PR.

## Regression Notes

1. Potential unintended areas of impact

This PR could introduce bugs in the plan purchase flow, but nothing outside of it.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I manually tested the flow as described above

3. What automated tests I added (or what prevented me from doing so)

The UI tests will be updated as mentioned above.

PR submission checklist:


UI Changes testing checklist:
- [x] ~~Languages with large words or with letters/accents not frequently used in English.~~ ⚠️ I only tested that text overflowed correctly, I don't expect issues with non-plain text because of built-in UTF8 support.
